dashboard/src/pages/[workspaceSlug]/[appSlug]/users/index.tsx

```diff
@@ -1,38 +1,326 @@
+import type { RemoteAppGetUsersQuery } from '@/utils/__generated__/graphql';
+import { useRemoteAppGetUsersQuery } from '@/utils/__generated__/graphql';
+import { SearchIcon } from '@heroicons/react/solid';
+import debounce from 'lodash.debounce';
+import { useRouter } from 'next/router';
+import type { ChangeEvent, ReactElement } from 'react';
+import { useEffect, useMemo, useRef, useState } from 'react';
+
+export type RemoteAppUser = Exclude<
+  RemoteAppGetUsersQuery['users'][0],
+  '__typename'
+>;
 
 export default function UsersPage() {
-  const { currentApplication } = useCurrentWorkspaceAndApplication();
+  const { openDialog, closeDialog } = useDialog();
+  const remoteProjectGQLClient = useRemoteApplicationGQLClient();
+  const [searchString, setSearchString] = useState<string>('');
+  const [currentPage, setCurrentPage] = useState(1);
+
+  const limit = useRef(25);
+  const [nrOfPages, setNrOfPages] = useState(1);
+  const router = useRouter();
+
+  const offset = useMemo(() => currentPage - 1, [currentPage]);
+
+  const remoteAppGetUserVariables = useMemo(
+    () => ({
+      where:
+        router.query.user !== undefined
+          ? {
+              id: {
+                _eq: searchString,
+              },
+            }
+          : {
+              _or: [
+                {
+                  displayName: {
+                    _ilike: `%${searchString}%`,
+                  },
+                },
+                {
+                  email: {
+                    _ilike: `%${searchString}%`,
+                  },
+                },
+              ],
+            },
+      limit: limit.current,
+      offset: offset * limit.current,
+    }),
+    [router.query.user, searchString, offset],
+  );
+
+  const {
+    data: dataRemoteAppUsers,
+    refetch: refetchProjectUsers,
+    loading: loadingRemoteAppUsersQuery,
+  } = useRemoteAppGetUsersQuery({
+    variables: remoteAppGetUserVariables,
+    client: remoteProjectGQLClient,
+  });
+
+  /**
+   * If a user of the app enters the users tab with a page query param of the following structure:
+   * `users?page=2` this useEffect will update the current page to 2.
+   * which in turn will update the offset and trigger fetching the data with the new variables.
+   * If the user enters a page number that is greater than the number of pages we will redirect
+   * the user to the first page and update the URL.
+   *
+   * @remarks If the user navigates the page back and forth we handle the URL change through
+   * props passed to the Pagination component.
+   * @see {@link Pagination}
+   *
+   */
+  useEffect(() => {
+    if (router.query.page && typeof router.query.page === 'string') {
+      const pageNumber = parseInt(router.query.page, 10);
+      if (nrOfPages >= pageNumber) {
+        setCurrentPage(pageNumber);
+      } else {
+        setCurrentPage(1);
+      }
+    }
+    // eslint-disable-next-line react-hooks/exhaustive-deps
```

Comment:
This line is here as updating `nrOfPages` was unnecessary running this side-effect.

I'll make a memoized value and delete it